tests/api/api-resources.test.js

```diff
@@ -0,0 +1,131 @@
+// Copyright Contributors to the Open Cluster Management project
+
+jest.retryTimes(global.retry)
+
+const { fail } = require('assert');
+const { performance } = require('perf_hooks');
+
+const squad = require('../../config').get('squadName')
+const {
+  getKubeConfig,
+  getSearchApiRoute,
+  getToken,
+  searchQueryBuilder,
+  sendRequest,
+} = require('../common-lib/clusterAccess')
+
+const {
+  fetchAPIResourcesWithListWatchMethods,
+  formatResourcesFromSearch,
+  formatFilters,
+  getResourcesFromOC,
+  getClusterList,
+  shouldUseAPIGroup
+} = require('../common-lib/index')
+
+const { sleep } = require('../common-lib/sleep')
+
+// Set list to ignore resources that aren't being collected by Search.
+const ignoreKindResourceList = ['clusterclaim', 'event']
+
+// Set list of resources that require filtering by api group.
+const requireAPIGroup = []
+
+/**
+ * Base test for kubernetes
+ * @param {*} kind
+ * @param {*} apigroup
+ * @param {*} cluster
+ * @param {*} user
+ * @param {*} namespace
+ */
+function baseTest(kind, apigroup, cluster = { type: 'hub', name: 'local-cluster' }, user = 'kubeadmin', namespace = '--all-namespaces') {
+  var runTest = test
+
+  try {
+    var expectedResources = getResourcesFromOC(kind, apigroup, namespace, cluster)
+  } catch (err) {
+    runTest = test.skip
+  }
+
+  runTest(
+    `[P2][Sev2][${squad}] verify data integrity for resource property: ${!apigroup.useAPIGroup ? kind : `${kind}.${apigroup.name}`}`,
+    async () => {
+      const filters = formatFilters(kind, apigroup, namespace, cluster)
+
+      // Fetch data from the search api.
+      var query = searchQueryBuilder({ filters })
+
+      var startTime = performance.now()
+      var resp = await sendRequest(query, token)
+      var endTime = performance.now()
+      var totalElapsedTime = endTime - startTime
+
+      var searchResources = formatResourcesFromSearch(resp)
+      
+      if (totalElapsedTime > 30000) {
+        fail(`Search required more than 30 second to return resources for ${kind}. (TotalElapsedTime: ${totalElapsedTime})`)
+      } else if (totalElapsedTime > 1000) {
+        console.warn(`Search required more than 1 second to return resources for ${kind}. (TotalElapsedTime: ${totalElapsedTime})`)
+      }
+
+      if (searchResources.length != expectedResources.length) {
+        console.warn(`Detected incorrect amount of data matches for (${kind}) resources. Retrying test within 5 seconds.`)
+        await sleep(5000)
+
+        // Refresh the list of resources. There's a chance that more resources were created after the previous fetch.
+        expectedResources = getResourcesFromOC(kind, apigroup, namespace, cluster)
+        resp = await sendRequest(query, token)
+        
```

Comment:
This retry is a good start but the test might still fail if there are changes in the cluster during the 5 second delay.
To make the test more robust, we'll need to find the mismatch resources and check only those in the retry.

I feel that the current implementation is good enough and will result in much less random errors than what we have today. So we can separate this improvement into a future PR. 

**Scenario 1:**
Search has more resources than CLI. After the retry we should see that the resources are either removed from search or added on the CLI.

**Scenario 2:**
Search is missing resources that are on the CLI. After the retry the missing resources should be added to search or removed from the CLI.

Reply:
For now, I think using the retry logic is okay at the moment. To prevent this PR from taking any longer, let's split the improvements into a future PR.